dodiscover/context_builder.py

```diff
@@ -0,0 +1,140 @@
+from typing import Optional, Set, Union
+
+import networkx as nx
+import pandas as pd
+
+from ._protocol import Graph
+from .context import Context
+
+
+class ContextBuilder:
+    """A builder class for creating Context objects ergonomically.
+
+    The context builder provides a way to capture assumptions, domain knowledge, and data.
+
+    Parameters
+    ----------
+    data : pd.DataFrame
+        A dataset in data-frame form, consisting of samples as rows and variables as columns.
+    """
+
+    _graph: Optional[Graph] = None
+    _included_edges: Optional[Union[nx.Graph, nx.DiGraph]] = None
+    _excluded_edges: Optional[Union[nx.Graph, nx.DiGraph]] = None
+    _observed_variables: Optional[Set[str]] = None
+    _latent_variables: Optional[Set[str]] = None
+
+    def __init__(self, data: pd.DataFrame) -> None:
+        self._data = data
```

Comment:
Let's not use data in context.  @adam2392 and @petergtz and others have convinced me to separate these out.  Others you and I have talked to recently @darthtrevino have convinced me as well.

Reply:
I acknowledge that separating data from context might impact @adam2392 's work but I think it would be a mild change.

Reply:
So the new API for algos should be: `algo.fit(data, context)`

?

Reply:
@adam2392 yep, latest changes reflect this